### PR TITLE
GS on Personal A/B: Unlock Custom Styles Modal

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to https://github.com/Automattic/dotcom-forge/issues/2862

### Premium Global Styles Upgrade Modal:
* Add support for the "**calypso_global_styles_personal**" experiment using the `useExperiment` hook from `calypso/lib/explat`.
* Introduce the `globalStylesOnPersonalExperiment` variable to determine whether the experiment variation is set to "**treatment**".
* Adjust the modal's content and messaging based on the `globalStylesOnPersonalExperiment` value, providing customized text for the Personal plan or the Premium plan.
* Update the "Included with your Premium plan" section to dynamically display features based on the experiment variation, showing different features for the Personal plan or the Premium plan.
* Ensure that the correct plan (Personal or Premium) is passed to the checkout page when the user clicks the upgrade button.

### Unified Design Picker Step:
* Implement support for the "calypso_global_styles_personal" experiment using the `useExperiment` hook.
* Define the `globalStylesOnPersonalExperiment` variable based on the experiment variation.
* Adapt the behavior of the "Checkout" button to set the plan as Personal or Premium, depending on the `globalStylesOnPersonalExperiment` value.

### Theme:
* Introduce the `globalStylesOnPersonalExperiment` variable using the `useExperiment` hook from 'calypso/lib/explat'.
* Adapt the behavior of the "Checkout" button to set the plan as Personal or Premium, depending on the `globalStylesOnPersonalExperiment` value.

### Theme Preview:
* Utilize the useExperiment hook to enable the "calypso_global_styles_personal" experiment and obtain the globalStylesOnPersonalExperiment value.
* Adapt the behavior of the "Checkout" button to set the plan as Personal or Premium, depending on the `globalStylesOnPersonalExperiment` value.

| Before | After |
--------|-------
image|Captura de Pantalla 2023-07-04 a las 9 14 35

### Flows
* Create a new site with a free plan, open the pattern assembler and change the fonts and/or colors during the creation process, once you press continue you should see the `Unlock Custom Styles Modal` pop up. `/setup/site-setup/patternAssembler?siteSlug={yoursite}&ref=site-selector`
* Create a new site with a free plan and change to a theme that has variations, select a paid variation and press continue, you should see the `Unlock Custom Styles Modal` pop up. `/setup/site-setup/designSetup?siteSlug={yoursite}&ref=site-selector&theme=george-lois&style_variation=dark`

### Tests
* Verify that the Premium Global Styles Upgrade Modal appears correctly and displays the appropriate content for the Personal and Premium plans based on the experiment variation.
* Test the functionality of the "Upgrade" button in the modal, ensuring that it redirects the user to the correct checkout page with the corresponding plan (Personal or Premium) selected.
* Overall, ensure that the affected components function as intended, maintaining the expected user experience while considering the changes related to the "calypso_global_styles_personal" experiment.

You can change your group here (21268-explat-experiment), you also need to clear the cache when switching between groups by executing this command in your sandbox:
`wp_cache_delete( 'global-styles-on-personal-{siteId}', 'a8c_experiments' );` 
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/premium-global-styles-upgrade-modal/index.tsx [hunk 1]
```diff
@@ -35,6 +38,13 @@ export default function PremiumGlobalStylesUpgradeModal( {
 		translate( 'Ad-free experience' ),
 		translate( 'Earn with WordAds' ),
 	];
+	const personalFeatures = [
+		<strong>{ translate( 'Free domain for one year' ) }</strong>,
+		translate( 'Style customization' ),
+		translate( 'Live chat support' ),
```
Comment (mmtr) [inline client/components/premium-global-styles-upgrade-modal/index.tsx]: The Personal plan does not have live chat support. Maybe we can highlight the gated/premium content feature?

Comment (Copons) [reply client/components/premium-global-styles-upgrade-modal/index.tsx]: If we want to keep the support feature, then: `Support via email` is what's on the pricing grid.

Comment (rcrdortiz) [reply client/components/premium-global-styles-upgrade-modal/index.tsx]: Done

**Review by dsas — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8350875

Thank you @rcrdortiz for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8350875

Thank you @rcrdortiz for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8350875

Thank you @rcrdortiz for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8350875).

**Pull request merged**